### ISSUE TITLE
Why output depth of all frames are of same scale in monocular videos?

### ISSUE THREAD

**Issue by biggiantpigeon** [nianticlabs/monodepth2#359]
I have a question after reading part **D.2. Single-Scale Evaluation** in the paper, when we use monocular videos, with scale ambiguity problem.
In the evaluation, you seem to assume the result of depth(and pose) of all frames(frame pairs) are the same. But in the loss function there is no constrains about this, so if different frames give their own depth in different scales, and pose accordingly, there will be no increase in the loss. So why the output scales of all frames are the same for a monocular video?
I don't know if I state my confusion correctly. Thank you in advance.

**Comment by mdfirman** [comment]
You are quite right – the model is, in theory, free to predict completely different pose and depth scales for each image. There is no loss telling it to make it the same for each frame.

However, it seems from our evaluation that the predicted depths are at least *reasonably* consistent between frames.

It kind of makes sense that scale would end up being fairly consistent. It's easier for the model to learn to make all cars which are 200pixels tall, for example, to the same depth – rather than having to learn a different mapping for each such car it sees in training data.

In my view this is sort of an Occum's Razor thing: assuming all the images in the training data contain pretty similar objects, it's simpler for the model to learn a reasonably scale consistent model, rather than different scales for each image.

Does this make some kind of sense to you?

**Comment by biggiantpigeon** [comment]
Thanks for your explaination. So does this mean that if I use my own training data, you don't recommend using videos from cameras with different intrinsics? (Unless in some other publications, intrinsic is also a learning parameter?)

**Comment by mdfirman** [comment]
Yes, I would avoid training data with very different intrinsics. Having some variation is probably fine, but not large changes.

**Comment by biggiantpigeon** [comment]
I see. Thanks a lot!
